azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -116,6 +121,17 @@ public void cleanUpStaleFlows() {
         Entry::getKey).forEach(this::cleanUpStaleFlows);
   }
 
+  /**
+   * Try to clean the stale containers that are older than maximum azkaban flow running time
+   */
+  public void cleanUpStaleContainers() {
```

Comment:
Note that any unchecked exceptions thrown by this method will potentially block subsequent executions of this method through in the cleanupService (quirk of the ScheduledExecutorService). 
Please catch and log any RuntimeException (do not rethrow). Look at `deleteContainerQuietly` for reference.

Reply:
@sshardool The scheduled executor instance we are using is a single-threaded one. we are adding two clean-up tasks in that (current one and earlier clean-up task). I believe it's okay given that our calls are not that heavy. Thinking out loud here.

Reply:
Thanks for pointing it out. I added RuntimeException catch block.

Reply:
Good point @svnarumugam, I was curious about it too and checked, seems like multiple schedules within a single ScheduledExecutor are supported, though not very obvious from the javadoc. Single thread is ok as the operations are exepected to be infrequent and quick.